module/zfs/zio.c

```diff
@@ -4441,6 +4431,15 @@ zio_vdev_io_start(zio_t *zio)
 			return (NULL);
 		}
 		zio->io_delay = gethrtime();
+
+		if (zio_handle_device_injection(vd, zio, ENOSYS) != 0) {
+			/*
+			 * "no-op" injections return success, but do no actual
+			 * work. Just return it.
+			 */
+			zio_interrupt(zio);
```

Comment:
I wonder if we ever want to delay those zios also, in which case it would probably need `zio_delay_interrupt()`.

Reply:
Yeah, why not! Changed, thanks.